### PR TITLE
Add custom node shapes

### LINKED ISSUE #43: Permanent node labels
## Elevator Pitch

Currently, `NodelLabels` are ephemeral (displayed upon hover). It would be useful to have a feature where all node labels could be displayed permanently at once. 

## Motivation

Seeing identifying information about many nodes at once helps interpret the graph.  

## Design Ideas

Image from https://studentwork.prattsi.org/infovis/visualization/visualizing-les-miserables-social-network-1862/:
 
image

### PR DESCRIPTION
**Pull request by nrbgt**
## References

- fixes #43

## Code changes

- [x] adds `Dynamic` classes for features
  - [x] add `coerce` for e.g. `number` or `boolean` types, applied per node/link
  - [x] subclasses 
    - [x] `Nunjucks` 
    - [x] `Column`
- [x] adds `NodeShape` behavior
- [x] adds `behaviors.shapes`
  - [x] `Text`
    - [x] expose most of the underlying features around text
  - [x] `BaseShape` (refactor of `Text`, one it works in 3d)
  - [x] `Ellipse`
  - [x] `Rectangle`
- [x] implement custom shapes in client
  - [x] 2d
  - [x] 3d
- [x] tests
- [ ] docs
  - [x] add example notebook
  - [ ] add demo widgets 

## User-facing changes

- users will be able to add stacks of shapes for nodes
- text
  - 2d
    - image
  - 3d 
    - text
      - image
    - ellipse
      - Screenshot from 2023-03-03 16-09-12
    - rectangle
      - Screenshot from 2023-03-03 16-08-52
 

## Backwards-incompatible changes

- this moves away from the `Behavior.(column_name|template)`, favoring dedicated widget subclasses
  - while only used in shapes at present, this should be expanded to the other behaviors for consistency, but would certainly force an `0.3.0` bump

## Future Work
- link shapes

### CODE REVIEW

**Comment by nrbgt** [comment]
Adding 3d will require provisioning additional libraries, etc. in the iframe in order to handle `instanceof` checks inside `ThreeJS`.

File: js/tokens.ts [hunk 1]
```diff
@@ -174,8 +187,20 @@ export type TSelectedSet = Set<string | number>;
...
+  (...args: any): string;
+}
+
 export type TCustomAction = 'reheat';
 
 export interface IActionMessage {
   action: TCustomAction;
 }
+
+export const FALSEY = ['', '0', 'false'];
```
Comment (sanbales) [inline js/tokens.ts:200]: should we add `null`, `[]`, and/or `{}`?

Comment (nrbgt) [reply js/tokens.ts]: 5cdca93

**Comment by nrbgt** [comment]
Still working on the demo stuff, but this is pretty much ready to go.

We could also push off fully fleshing out the demo notebooks to #58, where it would create a lot of conflicts anyway.

**Review by dfreeman06 — APPROVED** [review]

**Pull request merged**